design/recovery-internals.md

```diff
@@ -0,0 +1,164 @@
+# FDB Recovery Internals
+
+FDB uses recovery to handle various failures, such as hardware and network failures. When the current transaction system no longer works properly due to failures, recovery is automatically triggered to create a new generation of the transaction system. 
+
+This document explains at the high level how the recovery works in a single cluster. The audience of this document includes both FDB developers who want to have a basic understanding of the recovery process and database administrators who need to understand why a cluster fails to recover. This document does not discuss the complexity introduced to the recovery process by the multi-region configuration.
+
+## Background
+
+## `SeverDBInfo` data structure
+
+This data structure contains transient information which is broadcast to all workers for a database, permitting them to communicate with each other. It contains, for example, the interfaces for cluster controller (CC), master, ratekeeper, and resolver, and holds the log system's configuration.  It also has a vector of all storage server's interfaces. The definition of the data structure is in `SeverDBInfo.h`. The data structure is not available to the client. 
```

Comment:
> It also has a vector of all storage server's interfaces.

I don't think this is true? Are you referring to this line in the code:

```
	std::vector<std::pair<uint16_t,StorageServerInterface>> storageCaches;
```
?

If so, this is not a list of storages but rather of caches and this will be removed before the next release (and was not part of fdb 6.2). 

Also:
> The data structure is not available to the client. 

Is not quite true: only part of the data structure is available in the client. `ClientDBInfo` is part of `ServerDBInfo` and contains the list of proxies.

Reply:
Good catch! Yeah, you are right. I was confused at the `storageCaches`. :)

Reply:
The last sentence `The data structure is not available to the client.` is changed to `Only part of the data structure, such as `ClientDBInfo` that contains the list of proxies, is  available to the client. `